content/swift/concepts/subscripts/subscripts.md

```diff
@@ -0,0 +1,134 @@
+
+## Using Subscripts to Get and Set Values
+
+### Syntax
+
+The syntax below shows how a value can be retrieved from an array using subscript notation, by passing an integer as the `index` in square brackets after the array name. Dictionary values are retrieved similarly, by passing the `key` in square brackets after the dictionary name.
+
+```pseudo
+array[index]
+dictionary[key]
+```
+
+New values are set using the same notation:
+
+```pseudo
+array[index] = new value
+dictionary[key] = new value
+```
+
+### Examples
+
+The following example demonstrates how to set a new value in an array and then print it:
+
+```swift
+// Declare an array.
+var herbGarden = ["Dill", "Lavender", "Parsley", "Rosemary", "Sage", "Thyme"]
+
+// Set the element at position 0 to "Mint", replacing "Dill".
+herbGarden[0] = "Mint"
+
+// Print the element at position 0
+print(herbGarden[0])
+```
+
+This example will output the following:
+
+```shell
+Mint
+```
+
+This next example shows how to set a new value in a dictionary and print it:
+
+```swift
+// Declare a dictionary.
+var gardenInfo = ["name": "Happy Herb Garden", "type": "Herb", "length": "1m"]
+
+// Set the element with the key "length" to "2m", replacing "1m".
+gardenInfo["length"] = "2m"
+
+// Print the element with the key "length".
+print(gardenInfo["length"]!)
+```
+
+> **Note:** The "!" forces the optional String? value to unwrap
+
+This example will output the following:
+
+```shell
+2m
+```
+
+## Creating Subscripts
+
+### Syntax
+
+Subscripts are created using the `subscript` keyword followed by the input parameters (e.g. `index` or `key`) in brackets and then the return type.
+
+The syntax below is for a subscript that can get and set values. Code for retrieving the value is contained in a `get {}` wrapper and code for setting the value is contained in a `set {}` wrapper.
+
+```pseudo
+subscript(input parameters) -> return type {
+    get {
+        return the value here
+    }
+    set(newValue) {
+        set the new value here
+    }
+}
+```
+
+If a subscript is read-only, the code for retrieving the value can be entered directly inside the `subscript {}` wrapper, omitting the `get {}` wrapper.
+
+```pseudo
+subscript(input parameters) -> return type {
+        return the value here
+}
+```
+
+### Example
+
+The example below defines a read-only subscript within a [`struct`](https://www.codecademy.com/resources/docs/swift/structures) called `FlowerGarden`.
```

Comment:
this topic is already linked above, this link seems superfluous

Reply:
I removed this superfluous link.